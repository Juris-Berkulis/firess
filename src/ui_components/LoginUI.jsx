import React from 'react';
import { Link } from 'react-router-dom';

export const LoginUI = (props) => {
    return (
    <div>
        <form onSubmit={props.handleSubmit}>
        <p>Заполните форму ниже, чтобы войти в свою учетную запись.</p>
        <div>
            <input
            placeholder="Email"
            name="email"
            type="email"
            onChange={props.handleEmailChange}
            value={props.email}
            />
        </div>
        <div>
            <input
            placeholder="Password"
            name="password"
            type="password"
            onChange={props.handlePassChange}
            value={props.password}
            />
        </div>
        <div>
            {props.error && <p>{props.error}</p>}
            <button type="submit">Войти</button>
        </div>
        <hr />
        <p>
            <span>Нет аккаунта? </span>
            <Link to="/signup">Регистрация</Link>
        </p>
        </form>
    </div>
    )
};
