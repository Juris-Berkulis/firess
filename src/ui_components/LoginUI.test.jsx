import { render, screen, fireEvent, act } from "@testing-library/react"
import { LoginUI } from "./LoginUI.jsx";
import '@testing-library/jest-dom';
import { BrowserRouter } from "react-router-dom";

const handleSubmit = jest.fn();
const handleEmailChange = jest.fn();
const handlePassChange = jest.fn();

describe('LoginUI component', () => {
    it('LoginUI snapshot', () => {
        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
            classes={jest.fn()}
        /></BrowserRouter>);

        component.debug();
        expect(component).toMatchSnapshot();
    });

    it('LoginUI renders with a <button> and a "войти" text', () => {
        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
            classes={jest.fn()}
        /></BrowserRouter>);

        expect(component.getByText('Войти')).toBeInTheDocument();
        expect(component.getByRole('button')).toBeInTheDocument();
    });

    it('Was the function called?', () => {
        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
            classes={jest.fn()}
        /></BrowserRouter>);

        const submitBtn = component.queryByTestId('idBtnSubmitLogin');

        act(() => {
            fireEvent.click(submitBtn);
        });

        expect(handleSubmit).toBeCalled();
    });

    it('Text input fields have the text', () => {
        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='example@example.com' 
            password='12345678'
            classes={jest.fn()}
        /></BrowserRouter>);

        const fieldEmail = component.queryByTestId('idEmailLogin');
        const fieldPassword = component.queryByTestId('idPasswordLogin');

        expect(screen.queryByDisplayValue('example@example.com')).toBe(fieldEmail);
        expect(screen.queryByDisplayValue('12345678')).toBe(fieldPassword);
    });

    it('Error renders if it is', () => {
        const error = 'The password is invalid or the user does not have a password.';

        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error={error} 
            email='example@example.com' 
            password='12345678'
            classes={jest.fn()}
        /></BrowserRouter>);

        expect(component.getByText(error)).toBeInTheDocument();

        expect(component.getByTestId('idErrorLogin'));
    });

    it("Error don't renders if it isn't", () => {
        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error=''
            email='example@example.com' 
            password='12345678'
            classes={jest.fn()}
        /></BrowserRouter>);

        const errorItem = component.queryByTestId('idErrorLogin');
        expect(errorItem).toBeNull();
    });

    it('functions call if user is writing in text input fields', () => {
        const component = render(<BrowserRouter><LoginUI 
            handleSubmit={handleSubmit} 
            handleEmailChange={handleEmailChange} 
            handlePassChange={handlePassChange}
            error='' 
            email='' 
            password=''
            classes={jest.fn()}
        /></BrowserRouter>);

        const fieldEmail = component.queryByTestId('idEmailLogin');
        const fieldPassword = component.queryByTestId('idPasswordLogin');

        act(() => {
            fireEvent.change(fieldEmail, {
                target: {
                    value: 'login',
                }
            });
            fireEvent.change(fieldPassword, {
                target: {
                    value: 'password',
                }
            });
        });

        expect(handleEmailChange).toBeCalled();
        expect(handlePassChange).toBeCalled();
    });
});
