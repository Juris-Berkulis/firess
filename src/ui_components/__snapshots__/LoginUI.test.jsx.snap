// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginUI component LoginUI snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <form>
          <p>
            Заполните форму ниже, чтобы войти в свою учетную запись.
          </p>
          <div
            class="undefined undefined"
          >
            <input
              class="undefined undefined"
              data-testid="idEmailLogin"
              name="email"
              placeholder="Email"
              type="email"
              value="example@example.com"
            />
          </div>
          <div
            class="undefined undefined"
          >
            <input
              class="undefined undefined"
              data-testid="idPasswordLogin"
              name="password"
              placeholder="Password"
              type="password"
              value="12345678"
            />
          </div>
          <div
            class="undefined undefined"
          >
            
            <button
              data-testid="idBtnSubmitLogin"
              type="submit"
            >
              Войти
            </button>
          </div>
          <p>
            <span>
              Нет аккаунта? 
            </span>
            <a
              href=""
            >
              Регистрация
            </a>
          </p>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <form>
        <p>
          Заполните форму ниже, чтобы войти в свою учетную запись.
        </p>
        <div
          class="undefined undefined"
        >
          <input
            class="undefined undefined"
            data-testid="idEmailLogin"
            name="email"
            placeholder="Email"
            type="email"
            value="example@example.com"
          />
        </div>
        <div
          class="undefined undefined"
        >
          <input
            class="undefined undefined"
            data-testid="idPasswordLogin"
            name="password"
            placeholder="Password"
            type="password"
            value="12345678"
          />
        </div>
        <div
          class="undefined undefined"
        >
          
          <button
            data-testid="idBtnSubmitLogin"
            type="submit"
          >
            Войти
          </button>
        </div>
        <p>
          <span>
            Нет аккаунта? 
          </span>
          <a
            href=""
          >
            Регистрация
          </a>
        </p>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
